check-kdm-data.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import sys
+import requests
+import json
+import yaml
+from tempfile import TemporaryDirectory
+from subprocess import Popen
+
+
+class ReleaseDataMissing(Exception):
+    def __init__(self, dist, version):
+        self.dist = dist
+        self.version = version
+        super().__init__(f"Configuration for released {dist} version {version} is missing")
+
+
+class ReleaseDataChanged(Exception):
+    def __init__(self, dist, version):
+        self.dist = dist
+        self.version = version
+        super().__init__(f"Images for released {dist} version {version} have changed")
+
+
+class UnexpectedHelmChart(Exception):
+    def __init__(self, chart, version, repo, url):
+        self.chart = chart
+        self.version = version
+        self.repo = repo
+        self.url = url
+        super().__init__(f"Unexpected chart URL for {repo}/{chart}@{version}: {url}")
+
+
+def main(*releases):
+    dev_data = json.load(open('data/data.json'))
+
+    for release in releases:
+        release_data_url = f"https://releases.rancher.com/kontainer-driver-metadata/{release}/data.json"
+        response = requests.get(release_data_url)
+        response.raise_for_status()
+        release_data = response.json()
+
+        check_rke(dev_data, release_data)
+        check_k3s(dev_data, release_data)
+        check_rke2(dev_data, release_data)
+
+
+def check_rke(dev_data, release_data):
+    dist = 'Rancher k8s'
+    dev_images = dev_data.get('K8sVersionRKESystemImages', {})
+    release_images = release_data.get('K8sVersionRKESystemImages', {})
+
+    check_versions(dev_images, release_images, dist)
+
+
+def check_k3s(dev_data, release_data):
+    dist = 'k3s'
+    dev_versions = get_releases(dev_data, dist)
+    release_versions = get_releases(dev_data, dist)
+
+    check_versions(dev_versions, release_versions, dist)
+
+
+def check_rke2(dev_data, release_data):
+    dist = 'rke2'
+    dev_versions = get_releases(dev_data, dist)
+    release_versions = get_releases(dev_data, dist)
+
+    check_versions(dev_versions, release_versions, dist)
+    check_rke2_charts(dev_versions)
+
+
+def check_rke2_charts(versions):
```

Comment:
I'm mixed on whether or not we include this bit at all. Since it's not used, it seems wasteful. But at the same time it's nice for parity with the dev-v2.6 branch.

Reply:
Yeah, I'd prefer to keep the checks the the same across branches, even if some branches are missing data to check.

Reply:
Yeah that is probably for the best!